sdk/appconfiguration/Azure.Data.AppConfiguration/src/Pageable/ConditionalPageableImplementation.cs

```diff
@@ -0,0 +1,204 @@
+        {
+            var message = CreateMessage(conditions, pageSizeHint, nextLink);
+            if (message == null)
+            {
+                return null;
+            }
+
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope(_scopeName);
+            scope.Start();
+            try
+            {
+                _pipeline.Send(message, _cancellationToken);
+                return GetResponse(message);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        private async ValueTask<Response> GetNextResponseAsync(MatchConditions conditions, int? pageSizeHint, string nextLink, CancellationToken cancellationToken)
+        {
+            var message = CreateMessage(conditions, pageSizeHint, nextLink);
+            if (message == null)
+            {
+                return null;
+            }
+
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope(_scopeName);
+            scope.Start();
+            try
+            {
+                if (cancellationToken.CanBeCanceled && _cancellationToken.CanBeCanceled)
+                {
+                    using var cts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken, _cancellationToken);
+                    await _pipeline.SendAsync(message, cts.Token).ConfigureAwait(false);
+                }
+                else
+                {
+                    var ct = cancellationToken.CanBeCanceled ? cancellationToken : _cancellationToken;
+                    await _pipeline.SendAsync(message, ct).ConfigureAwait(false);
+                }
+
+                return GetResponse(message);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        private HttpMessage CreateMessage(MatchConditions conditions, int? pageSizeHint, string nextLink)
+        {
+            if (!string.IsNullOrEmpty(nextLink))
+            {
+                return _createNextPageRequest(conditions, pageSizeHint, nextLink);
+            }
+
+            return _createFirstPageRequest(conditions, pageSizeHint);
+        }
+
+        private Response GetResponse(HttpMessage message)
+        {
+            if (message.Response.IsError && _errorOptions != ErrorOptions.NoThrow)
+            {
+                throw new RequestFailedException(message.Response);
+            }
+
+            return message.Response;
+        }
+
+        private bool TryGetItemsFromResponse(Response response, out string nextLink, out List<ConfigurationSetting> items)
+        {
+            if (response is null)
+            {
+                nextLink = default;
+                items = default;
+                return false;
+            }
+
+            var parsedResponse = _responseParser(response);
```

Comment:
It cannot return null because it returns a non-nullable type.

It's not supposed to throw, unless the service breaks the swagger contract. In case the service returns an invalid response, I believe that throwing would be the best alternative as opposed to failing silently.